rts/Sim/Units/UnitTypes/Builder.cpp

```diff
@@ -731,11 +748,12 @@ bool CBuilder::StartBuild(BuildInfo& buildInfo, CFeature*& feature, bool& inWait
 
 				if (cu == nullptr)
 					continue;
+				if (allyteam != cu->allyteam)
+					return false; // Enemy units that block always block the cell
 				if (!CanAssistUnit(cu, buildInfo.def))
 					continue;
 
 				u = cu;
-				break;
```

Comment:
Since the loop now continues, `o` (see line 730/747) can refer to something other than `u` after the loop (for example if there is one `nullptr` in `cells` afterwards). On the other hand I don't see `o` being used so it might be a good moment to remove it.